controller/users.go

```diff
@@ -79,6 +80,41 @@ func (c *UsersController) Show(ctx *app.ShowUsersContext) error {
 	})
 }
 
+// UpdateUserAsServiceAccount updates a user when requested using a service account token
+func (c *UsersController) UpdateUserAsServiceAccount(ctx *app.UpdateUserAsServiceAccountUsersContext) error {
+
+	isSvcAccount, err := isServiceAccount(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to updated a user.")))
+	}
+
+	idString := ctx.ID
+	id, err := uuid.FromString(idString)
+	if err != nil {
+		// hide the implement details and return a NotFound code.
+		jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(errs.New("identity not found")))
```

Comment:
It's rather a bad request if ctx.ID is not a valid UUID

Reply:
I think..
the fact that it's not a valid UUID is an implementation detail - we should probably log the error, but to the external world , it should be considered as just another ID which wasn't found.

Reply:
Having said that I realize that we've been doing `BadRequest` everywhere, so I will go with consistency and change this to bad request :)

Reply:
Fixed https://github.com/fabric8-services/fabric8-wit/pull/1637/commits/fccb1b01c0b946bfeb2d8be8e4e948f06f7ac45f